theano/compile/tests/test_function_module.py

```diff
@@ -907,6 +908,34 @@ def test_empty_givens_updates():
     function([theano.In(x)], y, updates={})
 
 
+def test_sync():
+    if theano.config.device == 'cuda' and theano.gpuarray.pygpu_activated:
+        x = T.fmatrix('x')
+        w = theano.shared(np.random.rand(2000, 2000).astype('float32'), 'w')
+        b = theano.shared(np.zeros((2000)).astype('float32'), 'b')
+
+        y = T.dot(x, x) + b.dimshuffle('x', 0)
+
+        updates = [(w, w + T.dot(w, x) + T.dot(w, w))]
+
+        f = theano.function([x], y, updates=updates)
+        f.sync_shared()
+        g = theano.function([x], y, updates=updates)
+        x_ = np.random.rand(2000, 2000).astype('float32')
+        f(x_)
+        g(x_)
+        t_0 = time.time()
```

Comment:
I don't understand this timing. I would do something like:

t0 = ..
for i in ...
    f()
    f.sync_update()
t1 = ...
for i in range ...
   f()
t2 = ...
assert t1-t0 is bigger then t2-t1.

the sync_update should slow down computation. I would use a small range then 1000. Maybe 10 or 50.